components/filter/filter.js

```diff
@@ -389,9 +390,10 @@ class Filter extends FocusMixin(LocalizeCoreElement(RtlMixin(LitElement))) {
 
 		if (this._isDimensionEmpty(dimension)) {
 			return html`
-				<p class="d2l-filter-dimension-info-message d2l-body-small" role="alert">
```

Comment:
Before we had `role="alert"` here so that after the user opened an empty dimension it would immediately tell them it was empty, or below after a search finished loading it would say "5 search results" or whatever without having to navigate.  Does `d2l-empty-state-simple` have any announcing built in?  I'm not sure if adding `role="alert"` onto the component would also "just work", or if maybe wrapping it in a div with that would have the same results 🤔

Reply:
There's nothing built in, no. I will do some testing on this to see what is possible, good observation!

Reply:
Adding `role="alert"` directly to the empty state component does work but it was previously being placed within an `aria-live` region and was sometimes being read out twice, at least on VO. I fixed that and moved the `aria-live` region down to the inner list layer which fixes that.

But it led me to finding more screenreader issues. I'm getting duplicate screenreader calls when it reads all items in the list
e.g. `Fall Fall Winter Winter Spring Spring`

Also, when you are in a multi-dimensional filter and drilled into a dimension it lists the number of items in the list as the number of dimension-sets in the menu rather than the dimension-set-values in the list.
e.g. 'Menu 3 items' but there are 4 items in the list. This one might not have an easy solution since we are using a list within a menu.

Can you repro any of those on NVDA? I'll spin up a defect for these.

Reply:
You're seeing these defects on `main`, or after your changes? On `main`, I'm seeing:

> But it led me to finding more screenreader issues. I'm getting duplicate screenreader calls when it reads all items in the list
> e.g. Fall Fall Winter Winter Spring Spring
> 

So most of the time, I open it and get "Application. Art checkbox not checked." But you're right that sometimes, it starts reading the list items instead and does say "Art Art (Twelve). Astronomy Astronomy (two)." It's hard to get it to do it again though lol, is VoiceOver more consistent?

> Also, when you are in a multi-dimensional filter and drilled into a dimension it lists the number of items in the list as the number of dimension-sets in the menu rather than the dimension-set-values in the list.
> e.g. 'Menu 3 items' but there are 4 items in the list. This one might not have an easy solution since we are using a list within a menu.

Definitely not seeing that with NVDA. NVDA will properly say "Course, sub-menu 1 of 3. 1 role applied." and then when you drill in it doesn't talk about the menu again. It doesn't talk about counts at all, because our list items don't have that in `grid` mode (since they are no longer using the list roles). So seems like VO is talking about the parent when it shouldn't be 😟

Reply:
> Adding role="alert" directly to the empty state component does work but it was previously being placed within an aria-live region and was sometimes being read out twice, at least on VO. I fixed that and moved the aria-live region down to the inner list layer which fixes that.

I wonder if that's causing the duplicate list item reading as well. It _seems_ like I needed to add that as part of the empty state work: https://github.com/BrightspaceUI/core/pull/1432/files.  So if we don't actually need it to announce empty state for VO and NVDA, we could try removing it

Reply:
I see them on both `main` and this branch. 

I can consistently get the double announcing. If I open it, it will say "Application. Art checkbox not checked" and then start reading them afterwards. Also if you search something and then clear it, it will start reading all of the items again and they'll be duplicated.


> So seems like VO is talking about the parent when it shouldn't be 😟

Yeah.. it's not good :( And it's a bit inconsistent but sometimes I'm able to navigate the menu while i'm drilled into a dimension set.

Reply:
> So if we don't actually need it to announce empty state for VO and NVDA, we could try removing it

Yeah, using `role="alert"` creates an assertive `aria-live` region so you don't need to set one yourself. Removing it would also fix the double announcing defect

Reply:
> If I open it, it will say "Application. Art checkbox not checked" and then start reading them afterwards

Interesting, NVDA just stops reading them until you arrow to the next one, and then just reads that one.

> Yeah.. it's not good :( And it's a bit inconsistent but sometimes I'm able to navigate the menu while I'm drilled into a dimension set.

This might be a `hierarchical-view` issue. It sounds a bit like this: https://github.com/BrightspaceUI/core/pull/1272.  But worse, if it's an only VO issue 😟

Reply:
Thanks for helping me to investigate. I created a defect for this: [DE52188](https://rally1.rallydev.com/#/?detail=/defect/686309407849&fdp=true).

Reply:
So currently, NVDA + Chrome works fine, but Firefox just says nothing 😞 now. If I move the `role="alert"` to the container `d2l-empty-state-container` `div`s instead, it works in both.  Does that work in VoiceOver?

Reply:
Ah, it works in VO. Moving the role up to the container doesn't appear to break anything in VO so it should be fine to move that role up to the div container